l10n_br_nfse/models/document.py

```diff
@@ -0,0 +1,279 @@
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from requests import Session
+import logging
+
+from erpbrasil.assinatura import certificado as cert
+from erpbrasil.transmissao import TransmissaoSOAP
+from erpbrasil.base import misc
+from erpbrasil.edoc.nfse import NFSeFactory
+
+from odoo import api, fields, models, _
+from odoo.addons.l10n_br_fiscal.constants.fiscal import (
+    MODELO_FISCAL_NFSE,
+    SITUACAO_EDOC_AUTORIZADA,
+    TAX_FRAMEWORK_SIMPLES_ALL,
+)
+from ..constants.nfse import (
+    NFSE_ENVIRONMENTS,
+    OPERATION_NATURE,
+    RPS_TYPE,
+    TAXATION_SPECIAL_REGIME,
+    RECEPCIONAR_LOTE_RPS,
+    CONSULTAR_SITUACAO_LOTE_RPS,
+)
+
+from .res_company import PROCESSADOR
+
+_logger = logging.getLogger(__name__)
+
+
+def fiter_processador_edoc_nfse(record):
+    if (record.processador_edoc == PROCESSADOR and
+            record.document_type_id.code in [
+                MODELO_FISCAL_NFSE,
+            ]):
+        return True
+    return False
+
+
+def fiter_provedor(record):
+    if record.company_id.provedor_nfse:
+        return True
+    return False
+
+
+class Document(models.Model):
+
+    _inherit = 'l10n_br_fiscal.document'
+
+    edoc_error_message = fields.Text(
+        readonly=True,
+        copy=False,
+    )
+
+    rps_number = fields.Char(
+        string='RPS Number',
+        copy=False,
+        index=True,
+    )
+    rps_type = fields.Selection(
+        string='RPS Type',
+        selection=RPS_TYPE,
+        default='1',
+    )
+    operation_nature = fields.Selection(
+        string='Operation Nature',
+        selection=OPERATION_NATURE,
+        default='1',
+    )
+    taxation_special_regime = fields.Selection(
+        string='Taxation Special Regime',
+        selection=TAXATION_SPECIAL_REGIME,
+        default='1',
+    )
+    verify_code = fields.Char(
+        string='Verify Code',
+        readonly=True,
+    )
+    nfse_environment = fields.Selection(
+        selection=NFSE_ENVIRONMENTS,
+        string="NFSe Environment",
+        default=lambda self: self.env.user.company_id.nfse_environment,
+    )
+
+    @api.model
+    def create(self, values):
+        for record in self.filtered(fiter_processador_edoc_nfse):
+            if not values.get('date'):
+                values['date'] = record._date_server_format()
+
+            if values.get('company_id'):
+                company_obj = record.env['res.company']
+                company = company_obj.browse(values['company_id'])
+                if company.provedor_nfse:
+                    if values.get('document_serie_id') and \
+                            not values.get('rps_number'):
+                        serie_id = record.document_serie_id.browse(
+                            values['document_serie_id'])
+                        values['rps_number'] = serie_id.next_seq_number()
+                    values['number'] = None
+        return super(Document, self).create(values)
```

Comment:
Nao seria melhor ter esses valores pelo default_get (chamando super e pegando o company_id do super)?
Me parece que se vc criar os documentos fiscais pelo API (por examplo para sincronisar com algum conector), vc poderia querer ter esses campos antes de implementar alguma logica adicional. Da mesma forma, vc poderia querer eles na tela ao criar um novo documento fiscal. Caso for pelo default_get, talvez seria interessante verificar que o default_get do invoice ira chama-lo tb.

Reply:
Comentei com o @luismalta em reunião agora pouco que isto deve ser implementado no método document_number

Reply:
Sim eu vi que tiraram o default_get depois, ficou melhor. Ou ate botar um default=... campo por campo se precisar.